src/trio_util/_exceptions.py

```diff
@@ -148,7 +170,26 @@ def multi_error_defer_to(*privileged_types: Type[BaseException],
         priority_errors = errors_by_priority[min(errors_by_priority)]
         if strict and len(priority_errors) > 1:
             # multiple unique exception objects at the same priority
-            if propagate_multi_error:
+            if propagate_exc_group:
                 raise
-            raise RuntimeError('Unhandled trio.MultiError') from root_multi_error
+            raise RuntimeError('Unhandled ExceptionGroup') from root_exc_group
         raise priority_errors[0]
+
+
+def multi_error_defer_to(
+    *privileged_types: Type[BaseException],
+    propagate_multi_error: bool = True,
+    strict: bool = True,
+) -> ContextManager[None]:
+    """Deprecated alias for exceptgroup_defer_to()."""
+    import warnings
+    warnings.warn(
```

Comment:
The renaming and warning probably isn't worth it (I suspect my org is the only one using these functions).

I'm not sure it even makes sense to carry these exception utilities into the future.  They were introduced as a way to make a pre-ExceptionGroup world more manageable.

Reply:
> I didn't consider this only really being for pre-ExceptionGroup code. Might just close this then, if they should be effectively deprecated?

I should refine my statement:  they were introduced as a way to make lack of `except*` syntax more manageable.  Without the syntax, it's a burden for API users to deal with "exception Foo or Foo wrapped inside an exception group".  The filtering utils of MultiError and the exceptiongroup shim don't help with it.

Since projects don't always have freedom to use the newest python versions, these utilities still have a place.  But it's still probably the case that no one is using them or will miss them.